src/app/courses/courses-list/course-item/course-item.component.ts

```diff
@@ -0,0 +1,29 @@
+import { DatePipe } from '@angular/common';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+
+@Component({
+  selector: 'app-course-item',
+  templateUrl: './course-item.component.html',
+  styleUrls: ['./course-item.component.css']
+})
+export class CourseItemComponent implements OnInit {
+  @Input() title: string = '';
+  @Input() description: string = '';
+  @Input() duration: number | undefined;
```

Comment:
Usually you don`t need to write _undefined_ type even if the input variable will not be passed
The Input() is not obligatory
You can just leave the type _number_ and it would be an object add the _?_ sign in the code not to receive errors

Reply:
OK, great thing, have forgotten it since TypeScript courses. Thank you for it